examples/options/src/logging.cpp

```diff
@@ -0,0 +1,95 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024-2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/options/logging.hpp"
+
+#include "traccc/examples/utils/printable.hpp"
+
+// System include(s).
+#include <format>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+namespace {
+class verbosity_counter : public boost::program_options::typed_value<int> {
+    public:
+    verbosity_counter(int* store)
+        : boost::program_options::typed_value<int>(store) {
+        default_value(0);
+        zero_tokens();
+    }
+
+    virtual ~verbosity_counter() = default;
+
+    virtual void xparse(boost::any& store,
+                        const std::vector<std::string>&) const {
+        store = boost::any(++m_value);
+    }
+
+    private:
+    mutable int m_value{0};
+};
+}  // namespace
+
+namespace traccc::opts {
+
+logging::logging() : interface("Logging Options") {
+    m_desc.add_options()(
+        "verbose,v", new verbosity_counter(&m_verbosity_decr),
+        "Increase verbosity (can be specified multiple times)")(
+        "quiet,q", new verbosity_counter(&m_verbosity_incr),
+        "Decrease verbosity (can be specified multiple times)");
```

Comment:
Interesting idea. But I think a way over-complicated one.

You can choose to either just make it into an integer option (my preference), or a string option. Which would be trivially translated into `traccc::Logging::Level` values.

I personally would find an `--output-level=1` flag much more user friendly than having to write `--verbose --verbose`. 🤔

Reply:
The idea is that you write `-vv`, which is standard across a very broad range of CLI applications. Requiring the user to specify an integer level is much less ergonomic.

Reply:
I can tell you that in HEP applications specifying the output level is what we usually do. Be it ROOT, Athena, etc.  I know you're right that some projects in the wild use the `-vv` formalism. But the `-qq` one I've definitely not seen yet. 🤔 So I don't think that's such a strong argument for this formalism.

I was trying to find examples underlining my case in https://github.com/acts-project/acts/tree/main/Examples/Scripts/Python, but **none** of those scripts allow you to set the output level. 😕 At least none that I would've found so far...

Reply:
Well, I don't recall ever seeing something like `--output-level=X` in the wild, and I think there is a good reason for that:

1. The `-v` flag is so engrained in the minds of people using CLI applications that anything other than it feels foreign.
2. It is not clear what numerical verbosity levels mean; is 0 the INFO level? Is 0 the most verbose level? Is 0 the most quiet level? There is no rhyme or reason to it.
3. Writing the log levels out fully requires the user to know which levels are available, and this differs per application.

Reply:
:white_flag: Fine, let's use `-vv`. This command line interface is not super important in the long run after all.

But you could still simplify the logic here a bit. Could you not have `verbosity_counter` be able to both increment and decrement the counter, based on a second constructor argument? Then you could have just a single integer member variable, which would have the "correct value" by the time the command line parsing finishes.

Reply:
I tried really hard to make that work, but I have found that Boost program options starts to really break down if you try to have multiple flags set the same variable. I kept running into some edge cases that would break the situation. I agree it would be nicer to have a single variable for this; I'd be very open to any suggestions to make this work.